### PR TITLE
mtr-exporter: support specifying multiple jobs

### PR DESCRIPTION
**Pull request by jakubgs**
## Description of changes

This ability has been added in `0.3.0` release:
* https://github.com/mgumz/mtr-exporter/releases/tag/0.3.0
* https://github.com/NixOS/nixpkgs/pull/252667

To achieve this a config file is generated and symlinked at `/etc/mtr-exporter.conf`.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [x] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/modules/services/networking/mtr-exporter.nix
Comment (aanderse) [inline nixos/modules/services/networking/mtr-exporter.nix:46]: `types.nonEmptyListOf` can replace the assertion you have below.

**Review by aanderse — CHANGES_REQUESTED** [review]
Overall looks good to me. Just a few comments which hopefully are useful to you.

File: nixos/modules/services/networking/mtr-exporter.nix
Comment (jakubgs) [reply nixos/modules/services/networking/mtr-exporter.nix:16]: Neat.

File: nixos/modules/services/networking/mtr-exporter.nix
Comment (jakubgs) [reply nixos/modules/services/networking/mtr-exporter.nix:46]: Had no idea this existed. Thanks.

File: nixos/modules/services/networking/mtr-exporter.nix
Comment (jakubgs) [reply nixos/modules/services/networking/mtr-exporter.nix:100]: Although to be honest I kinda like keeping the `/etc` symlink for easier discoverability.

Comment (aanderse) [reply nixos/modules/services/networking/mtr-exporter.nix]: Sure. Many people do. Sounds good.

File: nixos/modules/services/networking/mtr-exporter.nix
Comment (aanderse) [reply nixos/modules/services/networking/mtr-exporter.nix:24]: Sorry about that... 😅

**Comment by jakubgs** [comment]
Thanks for the review. While applying feedback I also added support for `extraFlags`, since that's always a good idea.

**Comment by jakubgs** [comment]
Would appreciate a second review. I've addressed the comments.

File: nixos/modules/services/networking/mtr-exporter.nix
Comment (aanderse) [inline nixos/modules/services/networking/mtr-exporter.nix:38]: shouldn't we add `mkRemovedOptionModule` for this one too?

File: nixos/modules/services/networking/mtr-exporter.nix
Comment (aanderse) [inline nixos/modules/services/networking/mtr-exporter.nix:61]: If we exclude this we can also drop the `assertion`.

File: nixos/modules/services/networking/mtr-exporter.nix
Comment (jakubgs) [reply nixos/modules/services/networking/mtr-exporter.nix:38]: Ah, forgot about that. Done.

File: nixos/modules/services/networking/mtr-exporter.nix
Comment (jakubgs) [reply nixos/modules/services/networking/mtr-exporter.nix:61]: Good point. Fixed.

**Review by aanderse — APPROVED** [review]

**Comment by aanderse** [comment]
i'm happy to merge after you resolve the conflicts
sorry for the slow action on my part

**Comment by jakubgs** [comment]
No worries, appreciate the review. Rebased.

**Review by aanderse — APPROVED** [review]

**Pull request merged**
